packages/@lwc/ssr-compiler/src/compile-template/transformers/component/slotted-content.ts

```diff
@@ -155,7 +166,10 @@ function getLightSlottedContent(rootNodes: IrChildNode[], cxt: TransformerContex
                     // '' is the default slot name. Text nodes are always slotted into the default slot
                     const slotName =
                         node.type === 'Text' ? b.literal('') : bAttributeValue(node, 'slot');
+                    const { isSlotted } = cxt;
+                    cxt.isSlotted = ancestorIndices.length > 0 || node.type === 'Slot';
                     addLightDomSlotContent(slotName, [...ancestorIndices, i]);
+                    cxt.isSlotted = isSlotted;
```

Comment:
Setting before a function call and then reverting immediately after feels like we should just do the updates within the function.